File: front/api/queries/line/line.ts
import { factoryFragment, productFragment } from "../../fragments";
import gql from "graphql-tag";

export default gql`
  query line($id: ID!) {
    line(id: $id) {
      id
      name
      factoryId
      productId
      takt
      codeCount
      userCount
      factory {
        ...factoryFragment
      }
      product {
        ...productFragment
      }
    }
  }
  ${factoryFragment}
  ${productFragment}
`;

File: server/app/graphql/object_types/line.rb
class ObjectTypes::Line < Types::BaseObject
  graphql_name "Line"

  field :id, ID, null: false
  field :factory_id, ID, null: false
  field :factory, ObjectTypes::Factory, null: false
  field :product_id, ID, null: false
  field :product, ObjectTypes::Product, null: false
  field :name, String, null: false
  field :takt, Float, null: true
  field :code_count, Int, null: true
  field :user_count, Int, null: true
end
